src/ssh/sftp_client.cpp

```diff
@@ -215,6 +218,18 @@ bool SFTPClient::push_dir(const fs::path& source_path, const fs::path& target_pa
         }
     }
 
+    for (auto it = subdirectory_perms.crbegin(); it != subdirectory_perms.crend(); ++it)
```

Comment:
Is it important to have all the subdirs sorted? I see you start by the end, there must be a reason I am missing. But, if not, I'd use and `std::map` or `std::unordered_map` instead of a vector of pairs. WDYT?

Reply:
My reasoning was that the permissions should be set in the reverse order in which the directories were created, to ensure that the permissions of a parent directory don't disallow setting the permissions of a child directory. For example, when copying `parent/child`, the `parent` dir is created, then the `child` one. But maybe the permissions of `parent` are in such a way that they do not allow setting the permissions of `child`, so we must first set the permissions of `child` and only then the ones of `parent`.

Reply:
Good, thanks. Makes sense. I'm OK with the vector of pairs, then. I'm not worried about complexity because (i) we never search there and (ii) the vector will never be enormous. It was just a matter of readability.